sharepoint/templates/secrets.yaml

```diff
@@ -18,8 +18,10 @@ metadata:
   labels:
     {{- include "common.lib.labels" . | nindent 4 }}
 type: Opaque
+{{- if .Values.config.sharepointPfx }}
 data:
   privatekey.pfx: {{ .Values.config.sharepointPfx | b64enc }}
+{{- end }}
```

Comment:
i think you want to stick the if around the whole secret, otherwise it might throw errors for creating an empty secret

```
{{- if .Values.config.sharepointPfx }}
apiVersion: v1
kind: Secret
metadata:
  name: {{ include "common.lib.fullname" . }}-service-volume-config
  labels:
    {{- include "common.lib.labels" . | nindent 4 }}
type: Opaque
data:
  privatekey.pfx: {{ .Values.config.sharepointPfx | b64enc }}
---
{{- end }}
```